covid19-etl/etl/cityofchicago.py

```diff
@@ -0,0 +1,410 @@
+            else:
+                summary_group_demographics_value_dict["gender"] = gender[
+                    submitter_value
+                ]
+
+        else:
+            return
+
+        summary_group_demographics_submitter_id = derived_submitter_id(
+            summary_clinical_submitter_id,
+            "summary_clinical",
+            "summary_group_demographic",
+            age_group_dict,
+        )
+
+        if (
+            summary_group_demographics_submitter_id
+            not in self.summary_group_demographics
+        ):
+            self.summary_group_demographics[
+                summary_group_demographics_submitter_id
+            ] = summary_group_demographics_value_dict
+        self.summary_group_demographics[summary_group_demographics_submitter_id][
+            record_type_mapping[record_type]
+        ] = record_value
+
+    def parse_cityofchicago_file(self, start_date, end_date):
+        # parse original file into value to be passed in sheepdog
+        city_of_chicago_url = (
+            CITYOFCHICAGO_CDH_URL
+            + "?$where=lab_report_date between '"
+            + start_date
+            + "' and '"
+            + end_date
+            + "'"
+        )
+
+        with closing(requests.get(city_of_chicago_url, stream=True)) as r:
+            f = (line.decode("utf-8") for line in r.iter_lines())
+            reader = csv.reader(f, delimiter=",", quotechar='"')
+            self.headers = next(reader)
+
+            if self.headers[0] == "404: Not Found":
+                print("Unable to get file contents, received {}.".format(self.headers))
+                return
+
+            obtained_h = self.headers[: len(self.expected_csv_headers)]
+            assert (
+                obtained_h == self.expected_csv_headers
+            ), "CSV headers have changed (expected {}, got {}). We may need to update the ETL code".format(
+                self.expected_csv_headers, obtained_h
+            )
+
+            for row in reader:
+                self.parse_row(row)
+
+    def parse_row(self, row):
+
+        """
+        according to row mapping in the dataset, in each row
+        column 0 would be lab report date
+        column 1,2 and 3 would be total number of cases, deaths and hospitalization per day which would be used in summary_clinical
+        column 3 to rest would be for summary_group_demographics for Age group, race , gender and ethincity
+        Here we are ignoring records which doesn't have any lab report dates
+        """
+        if row[0] != "":
+            row_dict = {}
+            file_latest_date = self.get_unified_date_format(row[0])
+            row_dict[self.headers[0]] = file_latest_date
+            summary_clinical_submitter_id = self.add_summary_clinical(
+                self.convert_str_to_int(row[1]),
+                self.convert_str_to_int(row[2]),
+                self.convert_str_to_int(row[3]),
+                datetime.strptime(file_latest_date, "%Y-%m-%d"),
```

Comment:
I did that because the original date is the formate `"%Y-%m-%dT%H:%M:%S.%f"`  the row in summary clinical should  have formate of `"%Y-%m-%dT%H:%M:%S"` and the format for summary clinical submitter id should only have `"%Y-%m-%d"`. 

So the first conversion converts from original to the row format (removing %f) and the second conversion on line 344 is for the last conversion (removing time)

Let me know what do you think?

Reply:
Because of new commit, the line number changed for this but I would still like your review for this.

Reply:
oh i see, makes sense
but in that case i think you are losing the hour/min/sec/ms information by the time you add it the summary_clinical record 🤔

Reply:
Yes. This is the norms which I was given for summary group demographics and it didn't had any entry of H:M:S

type | project_id | submitter_id | age_group | race | gender | ethnicity | count | deaths | ***hospitalizations*** | summary_clinicals.submitter_id
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
summary_group_demographics | open-cityofchicago | summary_group_demographic_us_il_cook_2021-03-14_less_than_20_None_None | less than 20 |   |   |   | 35 | 0 | 0 | summary_clinical_us_il_chicago_2021-03-14

Reply:
oh okay. i think it's weird to enforce `date` as `%Y-%m-%dT%H:%M:%S` if we are removing the H:M:S information. In that case Y-M-D is enough. Wdym "the norms which I was given"? did someone from my team say we should do it this way? just curious where this is coming from

Reply:
Sean gave me this sample files on what this data should look like and I am following those.

Reply:
i don't see the `date` in the example row you shared

tbh i think it should either be "%Y-%m-%d" only, or include the exact time info we got from the source dataset

Reply:
So The date format `%Y-%m-%dT%H:%M:%S` is used for `summary_clinical`



type | project_id | submitter_id | count | deaths | hospitaliIzedCumulative | date | summary_locations.submitter_id
-- | -- | -- | -- | -- | -- | -- | --
summary_clinical | open-cityofchicago | summary_clinical_us_il_chicago_2021-03-14 | 185 | 2 | 10 | 2022-03-14T00:00:00.000 | summary_location_us_il_chicago

The date formate `"%Y-%m-%d"` is used for `summary_group_demographics` as mentioned above.

So I am taking `%Y-%m-%dT%H:%M:%S.%f` from input ,
converting it to `"%Y-%m-%d"` for summary_clinical_submitter_id and `%Y-%m-%dT%H:%M:%S` for date column,
and using ``%Y-%m-%dT%H:%M:%S` to get `"%Y-%m-%d"` for `summary_group_demographics_submitter_id`

Let me know what formats I should be using and I can make changes accordingly.

Reply:
Oh I just saw date format in summary_clinical have `%Y-%m-%dT%H:%M:%S.%f`. My bad. So I would make a commit to remove the conversion from `%Y-%m-%dT%H:%M:%S.%f` to `%Y-%m-%dT%H:%M:%S`

Reply:
I made a new commit with date formats changed.

Reply:
i don't know where you are seeing that, in the datasets i checked in prod, the `date` is in format YYYY-MM-DD. Can we use that everywhere in a consistent way? there is no need to store the time for daily data